src/coins.cpp

```diff
@@ -19,6 +23,41 @@ bool CCoinsView::HaveCoin(const COutPoint &outpoint) const
     return GetCoin(outpoint, coin);
 }
 
+bool CCoinsView::FindScriptPubKey(std::atomic<int>& scanProgress, std::atomic<bool>& shouldAbort, int64_t& count, CCoinsViewCursor& cursor, const std::set<CScript>& needles, std::map<COutPoint, Coin>& out_results) {
```

Comment:
Since we access the instance cursor (`CCoinsView()::Cursor()`), wouldn't it then require to access `pcoinsdbview` from within `coins.cpp` which seems not ideal?